cmd/swarmctl/node/common.go

```diff
@@ -138,6 +138,16 @@ func getNode(ctx context.Context, c api.ControlClient, input string) (*api.Node,
 		}
 
 		if l := len(rl.Nodes); l > 1 {
+			for _, n := range rl.Nodes {
+				name := n.Spec.Annotations.Name
```

Comment:
> The Docker doesn't check Annotations.Name.

This is only the current state of things. We have a name for nodes and we should use it. Operators should be able to name their nodes as they see fit, be it through hostname or through `NodeSpec` updates. We need to support both use cases.